### PR TITLE
feature/dashboard restaurant screen

### PR DESCRIPTION
**Pull request by mujtaba912**
implement Restaurant Screen in client dashboard

### CODE REVIEW

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/data/remote/mapper/RestaurantMapper.kt [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    restaurantName = restaurantName,
+    ownerUsername = ownerUsername,
+    phoneNumber = phoneNumber,
+    rating = rating,
+    priceLevel = priceLevel,
+    workingHours = workingHours
+)
+
+
+fun List<RestaurantDto>.toEntity() = map { it.toEntity() }
```
Comment (Saif-Alhaider) [inline client_dashboard/common/src/commonMain/kotlin/org/thechance/common/data/remote/mapper/RestaurantMapper.kt:17]:
```suggestion
fun List<RestaurantDto>.toEntity() = map(RestaurantDto::toEntity)
```

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/data/remote/model/RestaurantDto.kt [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package org.thechance.common.data.remote.model
+
+data class RestaurantDto(
+    val id: String,
+    val restaurantName: String,
```
Comment (Saif-Alhaider) [inline client_dashboard/common/src/commonMain/kotlin/org/thechance/common/data/remote/model/RestaurantDto.kt:5]:
```suggestion
    val name: String,
```
no need to mention the restaurant name in the variable the data class already mentions it

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/domain/entity/Restaurant.kt [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package org.thechance.common.domain.entity
+
+data class Restaurant(
+    val id: String,
+    val restaurantName: String,
+    val ownerUsername: String,
+    val phoneNumber: String,
+    val rating: Double,
+    val priceLevel: Int,
+    val workingHours: String,
```
Comment (Saif-Alhaider) [inline client_dashboard/common/src/commonMain/kotlin/org/thechance/common/domain/entity/Restaurant.kt:10]: working hours in the domain layer could be of type pair date

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/domain/entity/Restaurant.kt [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package org.thechance.common.domain.entity
+
+data class Restaurant(
+    val id: String,
+    val restaurantName: String,
```
Comment (Saif-Alhaider) [inline client_dashboard/common/src/commonMain/kotlin/org/thechance/common/domain/entity/Restaurant.kt:5]:
```suggestion
    val name: String,
```
no need to mention the restaurant name in the variable the data class already mentions it

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/domain/usecase/GetRestaurantsUseCase.kt [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+interface IGetRestaurantsUseCase {
+    suspend fun getRestaurants(): List<Restaurant>
+}
+
+class GetRestaurantsUseCase(private val remoteGateway: IRemoteGateway) : IGetRestaurantsUseCase {
+    override suspend fun getRestaurants(): List<Restaurant> {
+        return remoteGateway.getRestaurants()
+    }
+}
```
Comment (Saif-Alhaider) [inline client_dashboard/common/src/commonMain/kotlin/org/thechance/common/domain/usecase/GetRestaurantsUseCase.kt:14]: if its one function that is invoked 
use invoke operator instead

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/restaurant/RestaurantMapper.kt [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+        restaurantName = restaurantName,
+        ownerUsername = ownerUsername,
+        phoneNumber = phoneNumber,
+        rating = rating,
+        priceLevel = priceLevel,
+        workingHours = workingHours,
+    )
+
+
+fun List<Restaurant>.toUiState() = map { it.toUiState() }
```
Comment (Saif-Alhaider) [inline client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/restaurant/RestaurantMapper.kt:16]:
```suggestion
fun List<Restaurant>.toUiState() = map(Restaurant::toUiState)
```

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/restaurant/RestaurantScreenModel.kt [hunk 7]
```diff
@@ -1,10 +1,77 @@
...
 
+    init {
+        getRestaurant()
+    }
 
-}
\ No newline at end of file
+    private fun getRestaurant() {
+        tryToExecute(getRestaurants::getRestaurants, ::onGetRestaurantSuccessfully, ::onError)
+    }
```
Comment (Saif-Alhaider) [inline client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/restaurant/RestaurantScreenModel.kt:21]:
```suggestion
    private fun getRestaurants() {
        tryToExecute(getRestaurants::getRestaurants, ::onGetRestaurantSuccessfully, ::onError)
    }
```
the function name is reffereing to get a single restaurant

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/restaurant/RestaurantUIEffect.kt [hunk 8]
```diff
@@ -0,0 +1,3 @@
+package org.thechance.common.presentation.restaurant
+
+sealed interface RestaurantUIEffect {}
```
Comment (Saif-Alhaider) [inline client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/restaurant/RestaurantUIEffect.kt:3]: add to do comment

**Review by HaidyAbuGom3a — APPROVED** [review]

**Review by Asiasama710 — APPROVED** [review]

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/domain/entity/Restaurant.kt [hunk 9]
```diff
@@ -0,0 +1,11 @@
+package org.thechance.common.domain.entity
+
+data class Restaurant(
+    val id: String,
+    val name: String,
+    val ownerUsername: String,
+    val phoneNumber: String,
+    val rating: Double,
+    val priceLevel: Int,
+    val workingHours: String,
```
Comment (Saif-Alhaider) [inline client_dashboard/common/src/commonMain/kotlin/org/thechance/common/domain/entity/Restaurant.kt:10]: use date not string

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/restaurant/RestaurantUiState.kt [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+        Header("Rate", 3f),
+        Header("Price Level", 3f),
+        Header("Working Hours", 3f),
+        Header("", 1f),
+    ),
+    val numberOfRestaurants: Int = 0,
+    val search: String = "",
+    val isFilterDropdownMenuExpanded: Boolean = false,
+    val filterRating: Double = 0.0,
+    val filterPriceLevel: Int = 0,
```
Comment (Saif-Alhaider) [inline client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/restaurant/RestaurantUiState.kt:25]: seperate it in a another data class

**Pull request merged**